onnxruntime/contrib_ops/rocm/bert/layer_norm.cuh

```diff
@@ -109,8 +109,52 @@ __device__ inline void LayerNorm(
   }
 }
 
+
 template <typename T, int TPB, int ILP>
-__device__ inline void LayerNormSmall(const T* input_v, const hipcub::KeyValuePair<T, T>& thread_data,
+__device__ inline void LayerNormVec(
+    const hipcub::KeyValuePair<T, T>& thread_data, const int ld, const int offset, const T* beta,
+    const T* gamma, const T epsilon, T* output) {
+  // Assuming thread_data is already divided by ld
+  using VecT = aligned_vector<T, ILP>;
+  using BlockReduce = hipcub::BlockReduce<hipcub::KeyValuePair<T, T>, TPB>;
+  __shared__ typename BlockReduce::TempStorage temp_storage;
+  __shared__ T mu;      // mean
+  __shared__ T rsigma;  // 1 / std.dev.
+
+  KeyValuePairSum pair_sum;
+  const auto sum_kv = BlockReduce(temp_storage).Reduce(thread_data, pair_sum);
+
+  if (threadIdx.x == 0) {
+    mu = sum_kv.key;
+    rsigma = Rsqrt(sum_kv.value - mu * mu + epsilon);
+  }
+  __syncthreads();
+
+  if (ILP * threadIdx.x < ld) {
+    T beta_v[ILP], gamma_v[ILP], output_v[ILP];
+    VecT* gamma_val = reinterpret_cast<VecT*>(&gamma_v);
+    VecT* output_val = reinterpret_cast<VecT*>(&output_v);
+
+    for (int i = threadIdx.x * ILP; i < ld; i += TPB * ILP) {
+      int idx = offset + i;
+      if (beta != nullptr) {
+        VecT* beta_val = reinterpret_cast<VecT*>(&beta_v);
+        *beta_val = *reinterpret_cast<const VecT*>(&beta[i]);
+      }
+      *gamma_val = *reinterpret_cast<const VecT*>(&gamma[i]);
+      *output_val = *reinterpret_cast<const VecT*>(&output[idx]);
+#pragma unroll
+      for (int k = 0; k < ILP; k++) {
+        output_v[k] = (beta != nullptr) ? gamma_v[k] * (output_v[k] - mu) * rsigma + beta_v[k] :
+                                          gamma_v[k] * (output_v[k] - mu) * rsigma;
+      }
+      *(reinterpret_cast<VecT*>(&output[idx])) = *reinterpret_cast<VecT*>(&output_v[0]);
+    }
+  }
+}
+
+template <typename T, int TPB, int ILP>
+__device__ inline void LayerNormSmallVec(const T* input_v, const hipcub::KeyValuePair<T, T>& thread_data,
```

Comment:
What is the reason to add a new function in this file, can we reuse the existing function?